modules/wc-product-editor/assets/js/e-wc-product-editor.js

```diff
@@ -0,0 +1,63 @@
+import { WooHeaderItem } from '@woocommerce/admin-layout';
+import { useEntityId } from '@wordpress/core-data';
+import { Button } from '@wordpress/components';
+import { registerPlugin } from '@wordpress/plugins';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+function EditWithElementorButton() {
+	const productId = useEntityId( 'postType', 'product' );
+	const { saveEntityRecord } = useDispatch( 'core' );
+
+	const postStatus = useSelect( ( select ) => {
+		return select( 'core' ).getEditedEntityRecord( 'postType', 'product', productId )?.status;
+	}, [ productId ] );
+
+	const handleClick = () => {
+		if ( 'auto-draft' === postStatus ) {
+			saveEntityRecord( 'postType', 'product', {
+				id: productId,
+				name: `Elementor #${ productId }`,
+				status: 'draft',
```

Comment:
This probably works, but keep in mind that if the user has made any changes to the current product, those will be lost since this does not account for the edits in the entity record.

On the flip side, if you do include edits you'll also need to account for a potentially invalid state.

Reply:
@joshuatf I noticed this during manual testing, but then I looked at our current implementation with the classic product editor (not sure what you call it 😃), and it has the same problem. So I decided not to inflate the scope of this task. Thanks!